src/transformers/models/fastspeech2_conformer/modeling_fastspeech2_conformer.py

```diff
@@ -135,6 +135,11 @@ class FastSpeech2ConformerWithHifiGanOutput(FastSpeech2ConformerModelOutput):
 
     waveform: torch.FloatTensor = None
 
+    def __post_init__(self):
+        super().__post_init__()
+        # move waveform to the the beginning of the ordered dictionnary
+        self.move_to_end("waveform", last=False)
```

Comment:
We have to put `waveform` first to:
1. Be more aligned with the docstrings
2. the TTS pipeline takes the first output of a TTS model (otherwise the mel spectrogram would be the first output)

Reply:
I think both of these are bad reasons: 

1. The docstring should reflect the code, not the other way around 
2. The model shouldn't adapt for the pipeline (it shouldn't know about the pipeline at all). 

The best reason would be if this means it's consistent with all the other TTS models. However, this is potentially a breaking change. 

This would be clearer if the outputs are just added directly, rather than inheriting from another model output.

Reply:
Thanks for providing clear explanations! 

Consistency with the other TTS models is the real reason behind the 2nd reason I gave. Do you think this is a good enough reason for a breaking change? If so, I'll make `FastSpeech2ConformerWithHifiGanOutput` its own very class instead of inheriting, and I'll copy out every arguments